tests/host/common/user_interface.cpp

```diff
@@ -0,0 +1,458 @@
+*/
+
+#include <lwip/def.h>
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <ifaddrs.h>
+#include <netinet/in.h>
+#include <string.h>
+#include <arpa/inet.h>
+
+
+extern "C"
+{
+
+#include <user_interface.h>
+
+uint8 wifi_get_opmode(void)
+{
+	return STATION_MODE;
+}
+
+phy_mode_t wifi_get_phy_mode(void)
+{
+	return PHY_MODE_11N;
+}
+
+uint8 wifi_get_channel (void)
+{
+	return 1;
+}
+
+uint8 wifi_station_get_current_ap_id (void)
+{
+	return 0;
+}
+
+station_status_t wifi_station_get_connect_status (void)
+{
+	return STATION_GOT_IP;
+}
+
+uint8 wifi_station_get_auto_connect (void)
+{
+	return 1;
+}
+
+bool wifi_station_get_config (struct station_config *config)
+{
+	strcpy((char*)config->ssid, "emulated-ssid");
+	strcpy((char*)config->password, "emulated-ssid-password");
+	config->bssid_set = 0;
+	for (int i = 0; i < 6; i++)
+		config->bssid[i] = i;
+	config->threshold.rssi = 1;
+	config->threshold.authmode = AUTH_WPA_PSK;
+	config->open_and_wep_mode_disable = true;
+	return true;
+}
+
+void wifi_fpm_close(void)
+{
+}
+
+sint8 wifi_fpm_do_sleep (uint32 sleep_time_in_us)
+{
+	usleep(sleep_time_in_us);
+	return 1;
+}
+
+void wifi_fpm_do_wakeup (void)
+{
+}
+
+void wifi_fpm_open (void)
+{
+}
+
+void wifi_fpm_set_sleep_type (sleep_type_t type)
+{
+	(void)type;
+}
+
+uint32_t global_ipv4_netfmt = 0; // global binding
+
+bool wifi_get_ip_info (uint8 if_index, struct ip_info *info)
+{
+	struct ifaddrs * ifAddrStruct = NULL, * ifa = NULL;
+	uint32_t ipv4 = lwip_htonl(0x7f000001);
+	uint32_t mask = lwip_htonl(0xff000000);
+
+	if (getifaddrs(&ifAddrStruct) != 0)
+	{
+		perror("getifaddrs");
+		exit(EXIT_FAILURE);
+	}
+	for (ifa = ifAddrStruct; ifa != NULL; ifa = ifa->ifa_next)
+	{
+		if (   ifa->ifa_addr
+		    && ifa->ifa_addr->sa_family == AF_INET // ip_info is IPv4 only
+		   )
+		{
+			if (lwip_ntohl(*(uint32_t*)&((struct sockaddr_in*) ifa->ifa_netmask)->sin_addr) != 0xff000000)
+			{
+				if (ipv4 == lwip_htonl(0x7f000001))
+				{
+					// take the first by default
+					ipv4 = *(uint32_t*)&((struct sockaddr_in*)ifa->ifa_addr)->sin_addr;
+					mask = *(uint32_t*)&((struct sockaddr_in*)ifa->ifa_netmask)->sin_addr;
+				}
+				if (host_interface && strcmp(ifa->ifa_name, host_interface) == 0)
+				{
+					// .. or the one specified by user on cmdline
+					ipv4 = *(uint32_t*)&((struct sockaddr_in*)ifa->ifa_addr)->sin_addr;
+					mask = *(uint32_t*)&((struct sockaddr_in*)ifa->ifa_netmask)->sin_addr;
+					break;
+				}
+			}
+		}
+	}
+	if (ifAddrStruct != NULL)
+		freeifaddrs(ifAddrStruct);
+
+	(void)if_index;
+	//if (if_index != STATION_IF)
+	//	fprintf(stderr, "we are not AP");
+	
+	if (global_ipv4_netfmt == NO_GLOBAL_BINDING)
+		global_ipv4_netfmt = ipv4;
+
+	if (info)
+	{
+		info->ip.addr = ipv4;
+		info->netmask.addr = mask;
+		info->gw.addr = ipv4;
+	}
+
+	return true;
+}
+
+uint8 wifi_get_listen_interval (void)
+{
+	return 1;
+}
+
+bool wifi_get_macaddr(uint8 if_index, uint8 *macaddr)
+{
+	macaddr[0] = 0xde;
```

Comment:
Is this your MAC?  Did you want to put in an obviously fake one?

Reply:
`memcpy(macaddr, "ESPMOK");` ?

Reply:
It's deba7ab1e (at most 0x42 times) ;)